### PR TITLE
CIV-6250 defendant response defendant dashboard status

### PR DESCRIPTION
**Pull request by jeswanth-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-6250

### Change description ###
Dependent on Legal Rep ITP - Defendant response: Defendant dashboard status

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by jeswanth-hmcts** [comment]
PR is raised against the branch [CIV-6241-p3-defendant-dashboard-status](https://github.com/hmcts/civil-service/tree/CIV-6241-p3-defendant-dashboard-status) just for code review since I am using the some of the methods from that branch.

**Review by laperla11 — APPROVED** [review]

**Review by laperla11 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/claimstore/ClaimStoreService.java [hunk 1]
```diff
@@ -53,4 +55,11 @@ private LocalDateTime createAtToCreateDate(CmcClaim claim) {
 
         return createdDate;
     }
+
+    private LocalDateTime setIntentionToProceedDeadline(CmcClaim claim) {
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/service/claimstore/ClaimStoreService.java:59]: this is not a good method in my opinion. it states that it sets something but instead it is getting intention to proceed deadline from cmc. perhaps this method should be renamed to get and should reside in cmc claim. then you would call cmcClaim.getIntentonToProceedDeadline(); in the builder

Comment (jeswanth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/service/claimstore/ClaimStoreService.java]: changed to cmcClaim.getIntentonToProceedDeadline() , removed setIntentionToProceedDeadline method

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimMatcher.java [hunk 2]
```diff
@@ -219,9 +222,9 @@ public boolean isCourtReviewing() {
 
     @Override
     public boolean isClaimEnded() {
-        return !Objects.isNull(caseData.getApplicant1ProceedsWithClaimSpec())
+        return (!Objects.isNull(caseData.getApplicant1ProceedsWithClaimSpec())
```
[the remark this answers is not in the document]
Comment (jeswanth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimMatcher.java]: I hope it is an Objects.nonNull method refactored to the mentioned method

**Review by TatsyMaslova — DISMISSED** [review]

**Pull request closed without merging**